jpa/eclipselink.jpa.test/src/org/eclipse/persistence/testing/tests/jpa/advanced/PersistenceUnitProcessorTest.java

```diff
@@ -35,42 +35,140 @@
+    private static URLStreamHandler dummyHandler =
+        new URLStreamHandler() {
             @Override
             protected URLConnection openConnection(URL u) throws IOException {
                 return null;
             }
         };
 
+    private static void checkPURootSimple(
+        String inputUrl,
+        String expectedOutput
+    ) throws Exception {
         Assert.assertEquals(
-                "file:/C:/Oracle/Middleware/Oracle_Home/wlserver/samples/domains/mydomain/servers/myserver/tmp/_WL_user/eclipselink-advanced-model/y986my/eclipselink-advanced-model_ejb.jar",
-                PersistenceUnitProcessor.computePURootURL(
-                        new URL("zip", "", -1, "/C:/Oracle/Middleware/Oracle_Home/wlserver/samples/domains/mydomain/servers/myserver/tmp/_WL_user/eclipselink-advanced-model/y986my/eclipselink-advanced-model_ejb.jar!/META-INF/persistence.xml", dummyZipHandler),
-                        "META-INF/persistence.xml"
-                ).toString()
+            expectedOutput,
+            PersistenceUnitProcessor.computePURootURL(
+                new URL(inputUrl),
+                "META-INF/persistence.xml"
+            ).toString()
         );
-        
+    }
+
+    private static void checkPURootCustom(
+        String inputScheme,
+        String inputFile,
+        String expectedOutput
+    ) throws Exception {
         Assert.assertEquals(
-                "file:/C:/Program Files/Middleware/Oracle_Home/wlserver/samples/domains/mydomain/servers/myserver/tmp/_WL_user/eclipselink-advanced-model/y986my/eclipselink-advanced-model_ejb.jar", 
-                PersistenceUnitProcessor.computePURootURL(
-                        new URL("zip", "", -1, "/C:/Program Files/Middleware/Oracle_Home/wlserver/samples/domains/mydomain/servers/myserver/tmp/_WL_user/eclipselink-advanced-model/y986my/eclipselink-advanced-model_ejb.jar!/META-INF/persistence.xml", dummyZipHandler), 
-                        "META-INF/persistence.xml"
-                ).toString()
+            expectedOutput,
+            PersistenceUnitProcessor.computePURootURL(
+                new URL(inputScheme, "", -1, inputFile, dummyHandler),
+                "META-INF/persistence.xml"
+            ).toString()
         );
-        
-        Assert.assertEquals(
-                "file:/C:/Program.Files/Middleware/Oracle_Home/wlserver/samples/domains/mydomain/servers/myserver/tmp/_WL_user/eclipselink-advanced-model/y986my/eclipselink-advanced-model_ejb.jar", 
-                PersistenceUnitProcessor.computePURootURL(
-                        new URL("zip", "", -1, "/C:/Program.Files/Middleware/Oracle_Home/wlserver/samples/domains/mydomain/servers/myserver/tmp/_WL_user/eclipselink-advanced-model/y986my/eclipselink-advanced-model_ejb.jar!/META-INF/persistence.xml", dummyZipHandler), 
-                        "META-INF/persistence.xml"
-                ).toString()
+    }
+
+    public static void testComputePURootURLForZipFile() throws Exception {
```

Comment:
I dont think this needs to be static

Reply:
Coming from Scala, I have to say that it doesn't have to be non-static :smile: . Nevertheless, I've changed this to fit the project style.